### ISSUE TITLE
[bug?] 0.0.161 removes blank lines under indented import blocks

### ISSUE THREAD

**Issue by smackesey** [charliermarsh/ruff#1083]
Ruff 0.0.161 seemed to change the handling of blank lines around imports significantly. This block was sorted with `0.0.160`:

```python
if TYPE_CHECKING:
    from dagster._grpc.client import DagsterGrpcClient


def sync_get_external_execution_plan_grpc(
    api_client: "DagsterGrpcClient",
    # ...
```

But 0.0.161 turned it into this:

```python
if TYPE_CHECKING:
    from dagster._grpc.client import DagsterGrpcClient
def sync_get_external_execution_plan_grpc(
    api_client: "DagsterGrpcClient",
    # ...
```

This seemed to happen wherever an import block was indented-- was that intended?

**Comment by charliermarsh** [comment]
Not intended, will fix in a sec.

**Comment by charliermarsh** [comment]
@smackesey - Are you running from Dagster root? Would love to test against Dagster to make sure I'm not missing cases.

**Comment by smackesey** [comment]
Thanks for the quick fix! Yes I'm running `ruff --select=I001 --fix `git ls-files '*.py'` from the root.

**Comment by charliermarsh** [comment]
Okay cool. I may revert some of the newline stuff for now, or try to fix some of the other changes that you're seeing, but will cut a new release in a bit.

**Comment by charliermarsh** [comment]
(I was trying to mimic isort's behavior of inserting newlines when import blocks precede class and function definitions.)

**Comment by charliermarsh** [comment]
Oddly, isort doesn't seem to enforce that behavior for indented imports -- that is, it doesn't add a newline here:

```py
if True:
    import os
    def f():
        pass
```

But, I think it should? I.e., I think that should be corrected to:

```py
```py
if True:
    import os


    def f():
        pass
```

**Comment by charliermarsh** [comment]
I changed that behavior here (https://github.com/charliermarsh/ruff/pull/1088), which reduces the churn with isort but I think is less desirable behavior, but curious if you have a preference @smackesey.

**Comment by smackesey** [comment]
I agree that your suggested correction is better-- probably a bug in isort that it doesn't enforce that. Probably never caught because so many people follow `isort` with a formatter like `black` which will add those newlines.

**Comment by charliermarsh** [comment]
There's one case here that I'm handling differently than `isort`, which is that I'm changing this:

```py
import os

# Comment here.

def f():
    pass
```

...to:

```
import os


# Comment here.

def f():
    pass
```

Because the next AST node is a function, and we so we insert two lines after the import. I'd like to fix this, but it's kind of a tedious case so just a heads up as you may run into it.

**Comment by smackesey** [comment]
I just tried `0.0.162` and I think due to this extra newline behavior you mention above, Ruff's `I001` ends up incompatible with black. Here's an example:

After running `black`:

```
        from dagster._core.events import DagsterEventType
        from dagster._core.storage.event_log.base import EventRecordsFilter

        def _wrap_asset_fn(materialization_fn):
```

And after running `ruff --fix --select=I101`:

```
        from dagster._core.events import DagsterEventType
        from dagster._core.storage.event_log.base import EventRecordsFilter


        def _wrap_asset_fn(materialization_fn):
```

**Comment by charliermarsh** [comment]
Ahh interesting, ok. Is that only in indented blocks? Will check + fix.

**Comment by charliermarsh** [comment]
Ok yeah, looks like Black has different rules in indented blocks. Will fix.

**Comment by charliermarsh** [comment]
Ok, I believe this is fixed in [v0.0.163](https://github.com/charliermarsh/ruff/releases/tag/v0.0.163) (building now). I tested against Dagster, with Black, etc. That release also contains the D415 autofix, which I also tested against Dagster.
